docs/canonicalk8s/snap/howto/intermediate-ca.md

```diff
@@ -0,0 +1,60 @@
+# How to use intermediate CAs with Vault
+
+By default, {{product}} will generate self-signed CA certificates
+for the Kubernetes services.
+
+Follow this guide to prepare an intermediate Certificate Authority (CA) using
+[HashiCorp Vault] and then configure {{product}} to use the generated
+certificates.
+
+```{include} /_parts/common_vault_intermediate_ca.md
+```
+
+## Pass intermediate CA certificates to {{product}}
+
+The CA certificates can be specified through the [bootstrap configuration file]
+using the following fields:
+
+| Field                  | Description                |
+|------------------------|----------------------------|
+| ``ca-crt``             | API server CA certificate  |
+| ``ca-key``             | API server CA key          |
+| ``client-ca-crt``      | client CA certificate      |
+| ``client-ca-key``      | client CA key              |
+| ``front-proxy-ca-crt`` | Front Proxy CA certificate |
+| ``front-proxy-ca-key`` | Front Proxy CA key         |
+
+Let's prepare a bootstrap configuration using our newly generated intermediate
+CA certificate.
+
+```
+cat <<EOF > myca/bootstrap.yaml
+ca-crt: |
+$(cat myca/intermediate.crt | sed 's/^/    /g')
+ca-key: |
+$(cat myca/intermediate.key | sed 's/^/    /g')
+cluster-config:
+  network:
+    enabled: true
+  dns:
+    enabled: true
+  local-storage:
+    enabled: true
+EOF
+```
+
+Now bootstrap the cluster:
+
+```
+k8s bootstrap --file myca/bootstrap.yaml
+```
+
```

Comment:
I would add a line about what to expect here. For example after a few moments there should be a success message.